magellan-library/src/main/java/com/wealthfront/magellan/core/Step.kt

```diff
@@ -10,10 +10,11 @@ import androidx.annotation.VisibleForTesting
 import androidx.viewbinding.ViewBinding
 import com.wealthfront.magellan.lifecycle.LifecycleAwareComponent
 import com.wealthfront.magellan.lifecycle.lifecycleWithContext
+import com.wealthfront.magellan.view.ActionBarModifier
 
 abstract class Step<V : ViewBinding>(
   createBinding: (LayoutInflater) -> V
-) : Navigable, LifecycleAwareComponent() {
+) : Navigable, LifecycleAwareComponent(), ActionBarModifier {
```

Comment:
I worry that we're piling too much into the `Step` and `Journey` classes, when we're trying to practice composition over inheritance. Should `Step`s really be `ActionBarModifier`s, or should they have one instead?

Reply:
Remove the ActionBarModifier and make the onNavigate method have a nullable config. We will hide the actionbar when its null